src/v/raft/follower_queues.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "raft/group_configuration.h"
+#include "utils/mutex.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/core/semaphore.hh>
+
+#include <absl/container/node_hash_map.h>
+
+namespace raft {
+
+class follower_queues {
+public:
+    follower_queues(vnode, uint32_t);
+
+    ss::future<std::optional<ss::semaphore_units<>>>
+      get_append_entries_unit(vnode);
+
+    ss::future<> stop();
+
+    ss::future<> update_with_configuration(const group_configuration&);
```

Comment:
should we store std::unique_ptr<ss::semaphore> in the container? as it stands, it looks like it is unsafe to ever call update_with_configuration if there is an outstanding unit that has been acquired via get_append_entries_unit